api/test/h3-data/h3-risk-map.spec.ts

```diff
@@ -170,44 +193,113 @@ describe('H3 Data Module (e2e) - Risk map', () => {
       indicatorType: INDICATOR_TYPES.UNSUSTAINABLE_WATER_USE,
       fakeTable,
       fakeColumn,
     });
-    jest.spyOn(h3DataRepository, 'getWaterRiskMapByResolution');
 
     const response = await request(app.getHttpServer())
       .get(`/api/v1/h3/map/risk`)
       .query({
         indicatorId: indicator.id,
+        resolution: 1,
         year: 2020,
         materialId: material.id,
-        resolution: 6,
       });
+    expect(response.body.errors[0].title).toEqual(
+      `There is no H3 Data for Indicator with ID: ${indicator.id}`,
+    );
+  });
 
-    expect(h3DataRepository.getWaterRiskMapByResolution).toHaveBeenCalled();
-    expect(response.body.data).toEqual([
-      {
-        h: '861203a4fffffff',
-        v: 103,
-      },
-    ]);
-    expect(response.body.metadata).toEqual({
-      quantiles: [103, 103, 103, 103, 103, 103, 103],
-      unit: 'tonnes',
+  test('When I try to GET a Risk-Map with correct queries, but there is no H3 Data available for requested Material, then I should get a proper error message', async () => {
+    const unit = await createUnit();
+    const indicator = await createIndicator({
+      unit,
+      name: 'Indicator Name',
+    });
+    await h3DataMock(
+      fakeTable2,
+      fakeColumn2,
+      h3AlternativeIndicatorFixture,
+      indicator.id,
+    );
+    const material = await createMaterial({
+      name: 'Material Name',
+    });
+
+    const response = await request(app.getHttpServer())
+      .get(`/api/v1/h3/map/risk`)
+      .query({
+        materialId: material.id,
+        indicatorId: indicator.id,
+        resolution: 1,
+        year: 2020,
+      });
+
+    expect(response.body.errors[0].title).toEqual(
+      `There is no H3 Data for Material with ID: ${material.id}`,
+    );
+  });
+
+  test('When I get a calculated H3 Water Risk Map with the necessary input values, then I should get the h3 data (happy case). Different results for different resolutions expected', async () => {
+    const { material, indicator } = await createWorldForRiskMapGeneration({
+      indicatorType: INDICATOR_TYPES.UNSUSTAINABLE_WATER_USE,
+      fakeTable,
+      fakeColumn,
     });
+
+    await h3DataMock(
+      fakeTable2,
+      fakeColumn2,
+      h3AlternativeIndicatorFixture,
+      indicator.id,
+    );
+
+    jest.spyOn(h3DataRepository, 'getWaterRiskMapByResolution');
+
+    const responseRes6 = await request(app.getHttpServer())
+      .get(`/api/v1/h3/map/risk`)
+      .query({
+        indicatorId: indicator.id,
+        year: 2020,
+        materialId: material.id,
+        resolution: 6,
+      });
+
+    const responseRes3 = await request(app.getHttpServer())
+      .get(`/api/v1/h3/map/risk`)
+      .query({
+        indicatorId: indicator.id,
+        year: 2020,
+        materialId: material.id,
+        resolution: 3,
+      });
+
+    expect(h3DataRepository.getWaterRiskMapByResolution).toHaveBeenCalledTimes(
```

Comment:
IMO this sort of spy usage is redundant - if, for some reason, this method was not called, the result would differ, and your test would pick it up anyway. Plus, it ties your test to the implementation (if you implement the exact same feature, with a different method, your test would yield a false negative)

@alexeh I know you are a fan, so maybe you'd like to offer a different take on this?

Reply:
also: if we decide to keep it, we should validate the arguments

Reply:
contrary to what everyone might think, I do agree with @tiagojsag . So I implemented these spies in the risk-map test suite in the time that some calculations weren't different (they have been changing in time) so it was, in my opinion, a good idea to know that depending on requested type, X service was called. Also, my understanding of H3 stuff was (even more) low

Now that we have a more clear understanding, and clearly different calcs for each kind of risk-map, I do agree that this could be removed, or, validate the arguments that it has been called